src/orchard/inspect.clj

```diff
@@ -95,6 +95,18 @@
   (-> (clear inspector)
       (inspect-render value)))
 
+(defn next-page
+  "Jump to the next page when inspecting a paginated sequence/map. Does nothing
+  if already on the last page."
+  [inspector]
+  (inspect-render (update-in inspector [:current-page] inc)))
```

Comment:
Shouldn't be there some sort of checking before `inc`?

I believe that's the root of our issues.

Reply:
Is the check for not going beyond the last page?
That will be a good addition too but it's not related to the current issue IMO.

Reply:
The main issues I fixed here are
1) down fn was not going beyond the first page.
2) Handle the discrepancy between page-size and no: of items on the page when a hash-map is being inspected. This discrepancy IIUC existed since the separation of key/values was implemented [here](https://github.com/clojure-emacs/orchard/pull/24).

Reply:
I have fixed the new issues at the cost of removing the safety check in next-sibling to not go beyond the last element. 
I am not sure if it's possible to know the total size of the inspected paginated collection. 
This check might have to be removed anyways when supporting next-sibling for other collection types.

Reply:
Hmm, ideally we'd only merge things that improve some aspect without regressing on others.

Perhaps you can decompose this PR into the bits that you are most sure about?

(Admittedly some of these problems are pretty involved, which is why I left them pending)

Reply:
Not really in a hurry to get this merged. It's probably better to leave it here for a couple of days and think about it more.

I was trying to support next-sibling across pages as well which might not be possible with the same safety check without knowing the full collection size.(1ex would be lazy sequences)

Another fallback option is to support next-sibling within a page (with the same safety check) and use next-page to navigate across pages.

Reply:
Thanks for understanding and for the sustained efforts - appreciated!

If it makes things simpler, it would be acceptable to not support `next-sibling` on lazy collections at all. They're relatively infrequent (vs vectors, hashmaps) so I'd favor having something correct+complete over the most frequent use cases.

(an elegant way to not support something here is to simply re-render the same Inspector state)